app/Http/Controllers/SquareController.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable Generic.Formatting.SpaceBeforeCast.NoSpace
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\SquareCompleteRequest;
+use App\Models\DuesTransaction;
+use App\Models\Payment;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Square\Models\CreateCheckoutRequest;
+use Square\Models\CreateOrderRequest;
+use Square\Models\Money;
+use Square\Models\Order;
+use Square\Models\OrderLineItem;
+use Square\Models\OrderServiceCharge;
+use Square\Models\OrderServiceChargeCalculationPhase;
+use Square\Models\OrderState;
+use Square\SquareClient;
+
+class SquareController extends Controller
+{
+    private const PER_TRANSACTION_FEE = 30;  // cents
+    private const PERCENTAGE_FEE = 2.9;
+
+    public function payDues(Request $request)
+    {
+        $user = $request->user();
+
+        $transactionWithNoPayment = DuesTransaction::doesntHave('payment')
+            ->where('user_id', $user->id)
+            ->latest('updated_at')
+            ->first();
+
+        $transactionWithIncompletePayment = DuesTransaction::where('user_id', $user->id)
+            ->whereHas('package', static function (Builder $q): void {
+                $q->whereDate('effective_end', '>=', date('Y-m-d'));
+            })->whereHas('payment', static function (Builder $q): void {
+                $q->where('amount', 0.00);
+                $q->where('method', 'square');
+            })->first();
+
+        if (null !== $transactionWithIncompletePayment) {
+            $transaction = $transactionWithIncompletePayment;
+
+            $payment = $transaction->payment[0];
```

Comment:
The previous implementation reset the `unique_id` here, which would allow for creating duplicate orders. This implementation may result in the `createCheckout` call failing if there is a long period between the first and second times someone attempts payment, but I think that should be ok. We can delete the payment if we need to.

Reply:
We could add a listener for order events and then clean up (how?) payments that would no longer work.